code/controllers/subsystem/weeds.dm

```diff
@@ -100,3 +102,18 @@ SUBSYSTEM_DEF(weeds)
 			swapped = TRUE
 			qdel(O)
 	new weed_to_spawn(T, node, swapped)
+
+/datum/controller/subsystem/weeds/proc/create_node(turf/T, obj/alien/weeds/node/node)
+	if(QDELETED(node) || (node.name != WEED) || !T.is_weedable() || iswallturf(T) || (locate(/obj/alien/weeds/node) in range(1, T)))
+		return
+	var/obj/alien/weeds/node/node_to_spawn = /obj/alien/weeds/node
```

Comment:
Uh, what is that code doing ?

Reply:
Checks if we are allowed to spawn a node

Reply:
I need to assign a new obj, otherwise it throws runtime errors

Reply:
It is not an object this is a typepath. You are creating this var then using it last line of function. This var is not usefull, just create a new node at the end of the function directly